### ISSUE TITLE
The downloaded file will not be moved.

### ISSUE THREAD

**Issue by uptoratlen** [uptoratlen/gsArchivPDFDownloader#6]
There must be a small error. The downloaded file will not be moved.

```
d:\Downloads\geckodriver-v0.29.0-win64\gsArchivPDFDownloader-0.5b>python gsArchivPDFDownloader.py -l
2021-04-30 01:03:20:[INFO ] gsArchivPDFDownloader Version:0.5b
2021-04-30 01:03:20:[INFO ] Download location:C:\Daten\Gamestar-Archive
2021-04-30 01:03:20:[INFO ] filenamepattern_fromserver:GameStar Nr. <ausgabe>_<jahr>.pdf
2021-04-30 01:03:20:[INFO ] filenamepattern_intarget  :GameStar_Nr._<ausgabe>_<jahr>.pdf
2021-04-30 01:03:26:[INFO ] Browser now started with URL:https://www.gamestar.de/plus/ - try now to log in with user/password [xxxxxxxxxxxxxxxxxxx/************************]
2021-04-30 01:03:28:[INFO ] Trying now to download the latest version for (Editions/Year) (curMonth/curYear)=>(4/2021) (4/2021)
2021-04-30 01:03:28:[INFO ] Process Download for 'C:\Daten\Gamestar-Archive/2021/GameStar Nr. 04_2021.pdf'
2021-04-30 01:03:33:[INFO ] Try now download of : Jahr 2021 and Ausgabe 4 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=4
2021-04-30 01:03:48:[INFO ] Download done successful
2021-04-30 01:03:50:[ERROR] Exception:[WinError 2] Das System kann die angegebene Datei nicht finden: 'C:\\Daten\\Gamestar-Archive/GameStar_Nr._4_2021.pdf' -> 'C:\\Daten\\Gamestar-Archive/2021/GameStar Nr. 04_2021.pdf'
Traceback (most recent call last):
  File "d:\Downloads\geckodriver-v0.29.0-win64\gsArchivPDFDownloader-0.5b\gsArchivPDFDownloader.py", line 145, in download_edition
    os.rename(f"{user_data[0]['downloadtarget']}/{filenamepattern_download}",
FileNotFoundError: [WinError 2] Das System kann die angegebene Datei nicht finden: 'C:\\Daten\\Gamestar-Archive/GameStar_Nr._4_2021.pdf' -> 'C:\\Daten\\Gamestar-Archive/2021/GameStar Nr. 04_2021.pdf'

```

_Originally posted by @hightower5 in https://github.com/uptoratlen/gsArchivPDFDownloader/issues/5#issuecomment-829654664_

**Comment by uptoratlen** [comment]
Let me check my test systems. I'm currently adding some unit tests and also break the large functions down to smaller ones. 
The issue reported is not that unfamiliar, but let me check. Maybe it is a based on config thing.

Could you:
a) set the log level to "debug" in gs.json and rerun
b) check if any file exists in your "C:\Daten\Gamestar-Archive" folder

Even since the very first version there is a comment at that lines:
"# Give it time to sync to disk - not clear"
So I saw this also way back. I will try to get it more reliable.

**Comment by hightower5** [comment]
to a) not a good idea. The log files are cryptic and endless.
to b) there are folders from 1997-2019 and 2021. These folders contain PDF-files like "GameStar_Nr._01_2021.pdf".

I have to stop the script manual with CTRL+C.

Here is my config of gs.json:

```
[
    {
        "user": "xxxxxx",
        "password": "******",
        "log_level": "DEBUG",
        "downloadtarget": "C:\\Daten\\Gamestar-Archive",
        "edition2d": "Yes",
        "downloadtimeout": 240,
        "filenamepattern_intarget":"GameStar_Nr._<ausgabe>_<jahr>.pdf",
        "filenamepattern_fromserver": "GameStar Nr. <ausgabe>_<jahr>.pdf",
        "latestdownload": [
            {
                "year": "2021",
                "edition": "3"
            }
        ],
        "browser_display_on_latest": "no",
        "editions": [
        ....
```

**Comment by hightower5** [comment]
I have tested the version 0.5b again.

With the default setting it works.
Problem makes my setting (` _Nr._ `): 
` "filenamepattern_intarget":"GameStar_Nr._<ausgabe>_<jahr>.pdf"`

**Comment by uptoratlen** [comment]
Confirm - Bug no also seen here.  Thx for the investigations done.

**Comment by uptoratlen** [comment]
Please try version 0.5.5. The issue was caused by using arguments in wrong order.  The infinite loop was caused by my assumption that at a certain time the file would exist. But if I check for the wrong file name, it will never been seen. This is now also solved (as it may not happen again, by a timeout).
But I have to give credit to you for drilling down the cause. Nice job. Now I cross fingers......

**Comment by hightower5** [comment]
Hi,

it seems that the latest version 0.5.5 works very well.

I've also tried to download the year 2021 complete (argument: -y 2021).  There wasn't any errors, but the download took longer than necessary, because the script tries to download all next month from june to december.

I've two questions: 

- what means "GameStar Nr. 5_2021.pdf.part Seen- waiting"? Because there are several lines with this text.
- is it possible to stop the download, when more than 2 pages was not found?

Log file:

```
python gsArchivPDFDownloader.py -y 2021
2021-05-02 12:52:26:[INFO ] gsArchivPDFDownloader Version:0.5.5
2021-05-02 12:52:26:[INFO ] Download location:C:\Daten\Gamestar-Archive
2021-05-02 12:52:26:[INFO ] filenamepattern_fromserver:GameStar Nr. <ausgabe>_<jahr>.pdf
2021-05-02 12:52:26:[INFO ] filenamepattern_intarget  :GameStar_Nr._<ausgabe>_<jahr>.pdf
2021-05-02 12:52:33:[INFO ] Browser now started with URL:https://www.gamestar.de/plus/ - try now to log in with user/password [xxxxxxxxxxxxxxxxxxx/************************]
2021-05-02 12:52:35:[INFO ] Filename after Download on disk :[GameStar Nr. 1_2021.pdf]
2021-05-02 12:52:35:[INFO ] Filename for Target(modified)   :[GameStar_Nr._01_2021.pdf]
2021-05-02 12:52:35:[INFO ] Skip download - already existing 'C:\Daten\Gamestar-Archive/2021/GameStar_Nr._01_2021.pdf'
2021-05-02 12:52:35:[INFO ] Filename after Download on disk :[GameStar Nr. 2_2021.pdf]
2021-05-02 12:52:35:[INFO ] Filename for Target(modified)   :[GameStar_Nr._02_2021.pdf]
2021-05-02 12:52:35:[INFO ] Skip download - already existing 'C:\Daten\Gamestar-Archive/2021/GameStar_Nr._02_2021.pdf'
2021-05-02 12:52:35:[INFO ] Filename after Download on disk :[GameStar Nr. 3_2021.pdf]
2021-05-02 12:52:35:[INFO ] Filename for Target(modified)   :[GameStar_Nr._03_2021.pdf]
2021-05-02 12:52:35:[INFO ] Skip download - already existing 'C:\Daten\Gamestar-Archive/2021/GameStar_Nr._03_2021.pdf'
2021-05-02 12:52:35:[INFO ] Filename after Download on disk :[GameStar Nr. 4_2021.pdf]
2021-05-02 12:52:35:[INFO ] Filename for Target(modified)   :[GameStar_Nr._04_2021.pdf]
2021-05-02 12:52:35:[INFO ] Skip download - already existing 'C:\Daten\Gamestar-Archive/2021/GameStar_Nr._04_2021.pdf'
2021-05-02 12:52:35:[INFO ] Filename after Download on disk :[GameStar Nr. 5_2021.pdf]
2021-05-02 12:52:35:[INFO ] Filename for Target(modified)   :[GameStar_Nr._05_2021.pdf]
2021-05-02 12:52:40:[INFO ] Try now download of : Jahr 2021 and Ausgabe 5 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=5
2021-05-02 12:52:52:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:52:54:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:52:55:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:52:57:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:52:58:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:00:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:01:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:03:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:04:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:06:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:07:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:09:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting
2021-05-02 12:53:10:[INFO ] Download done successful
2021-05-02 12:53:10:[INFO ] Trying now to move the downloaded file with a timeout of:[30]
2021-05-02 12:53:10:[INFO ] Move Download done successful
2021-05-02 12:53:10:[INFO ] Filename after Download on disk :[GameStar Nr. 6_2021.pdf]
2021-05-02 12:53:10:[INFO ] Filename for Target(modified)   :[GameStar_Nr._06_2021.pdf]
2021-05-02 12:53:15:[INFO ] Try now download of : Jahr 2021 and Ausgabe 6 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=6
2021-05-02 12:53:34:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:53:34:[INFO ] Filename after Download on disk :[GameStar Nr. 7_2021.pdf]
2021-05-02 12:53:34:[INFO ] Filename for Target(modified)   :[GameStar_Nr._07_2021.pdf]
2021-05-02 12:53:39:[INFO ] Try now download of : Jahr 2021 and Ausgabe 7 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=7
2021-05-02 12:53:57:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:53:57:[INFO ] Filename after Download on disk :[GameStar Nr. 8_2021.pdf]
2021-05-02 12:53:57:[INFO ] Filename for Target(modified)   :[GameStar_Nr._08_2021.pdf]
2021-05-02 12:54:02:[INFO ] Try now download of : Jahr 2021 and Ausgabe 8 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=8
2021-05-02 12:54:20:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:54:20:[INFO ] Filename after Download on disk :[GameStar Nr. 9_2021.pdf]
2021-05-02 12:54:20:[INFO ] Filename for Target(modified)   :[GameStar_Nr._09_2021.pdf]
2021-05-02 12:54:25:[INFO ] Try now download of : Jahr 2021 and Ausgabe 9 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=9
2021-05-02 12:54:44:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:54:44:[INFO ] Filename after Download on disk :[GameStar Nr. 10_2021.pdf]
2021-05-02 12:54:44:[INFO ] Filename for Target(modified)   :[GameStar_Nr._10_2021.pdf]
2021-05-02 12:54:49:[INFO ] Try now download of : Jahr 2021 and Ausgabe 10 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=10
2021-05-02 12:55:07:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:55:07:[INFO ] Filename after Download on disk :[GameStar Nr. 11_2021.pdf]
2021-05-02 12:55:07:[INFO ] Filename for Target(modified)   :[GameStar_Nr._11_2021.pdf]
2021-05-02 12:55:12:[INFO ] Try now download of : Jahr 2021 and Ausgabe 11 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=11
2021-05-02 12:55:30:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:55:30:[INFO ] Filename after Download on disk :[GameStar Nr. 12_2021.pdf]
2021-05-02 12:55:30:[INFO ] Filename for Target(modified)   :[GameStar_Nr._12_2021.pdf]
2021-05-02 12:55:35:[INFO ] Try now download of : Jahr 2021 and Ausgabe 12 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=12
2021-05-02 12:55:53:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:55:53:[INFO ] Filename after Download on disk :[GameStar Nr. 13_2021.pdf]
2021-05-02 12:55:53:[INFO ] Filename for Target(modified)   :[GameStar_Nr._13_2021.pdf]
2021-05-02 12:55:58:[INFO ] Try now download of : Jahr 2021 and Ausgabe 13 - URL:https://www.gamestar.de/_misc/plus/showbk.cfm?bky=2021&bkm=13
2021-05-02 12:56:17:[WARNI] Looks like the page not found is displayed - skip this edition
2021-05-02 12:56:17:[INFO ] Last requested edition downloaded - give job some time (30s) to finish, for no good reason...
2021-05-02 12:56:47:[INFO ] Job done
```
Best regards

**Comment by uptoratlen** [comment]
Thx, for testing. Will close this one as accepted. 
The log line .
"2021-05-02 12:52:52:[INFO ] C:\Daten\Gamestar-Archive/GameStar Nr. 5_2021.pdf.part Seen- waiting"

means, the download is in progress. I changed this log line from DEBUG to INFO, as I thought this might be a typical issue. In fact it was not. I will revert this line to be DEBUG again. The edition 5 is very large in fact.

The other question was also in my mind, but as you requested it also. I made it a enhancement for the time when I/we fixed the current issue #7.
